pkg/email/send_email.go

```diff
@@ -11,30 +14,35 @@ type Emailer struct {
 	Limiter *rate.Limiter
 }
 
+var (
+	ErrRateLimitExceeded = errors.New("rate limit exceeded")
+)
+
 const (
 	defaultRateLimit = 1.0 / 60.0
 	defaultBurstRate = 10
 )
 
 type EmailParams struct {
-	SMTPHost  string
-	Port      int
-	Username  string
-	Password  string
-	Sender    string
-	RateLimit float64
-	Burst     int
+	SMTPHost   string
+	Port       int
+	Username   string
+	Password   string
+	Sender     string
+	LimitEvery float64
+	Burst      int
 }
 
 func NewEmailer(e EmailParams) Emailer {
 	d := gomail.NewDialer(e.SMTPHost, e.Port, e.Username, e.Password)
-	if e.RateLimit == 0 {
-		e.RateLimit = defaultRateLimit
+	if e.LimitEvery == 0 {
+		e.LimitEvery = defaultRateLimit
 	}
 	if e.Burst == 0 {
 		e.Burst = defaultBurstRate
 	}
-	l := rate.NewLimiter(rate.Limit(e.RateLimit), e.Burst)
+	limit := float64(time.Second) / float64(e.LimitEvery)
```

Comment:
check the linter error

Reply:
The warning is because `LimitEvery` is still not a `Duration`.  But: this place is where @nopcoder suggested simply using the `Every` method!

Reply:
Used indeed! Thanks